core/embed/sec/tropic/tropic.c

```diff
@@ -395,8 +423,82 @@ static void lt_r_mem_data_write_time(uint32_t *time_ms) {
   *time_ms += 77;
 }
 
+static void lt_mcounter_get_time(uint32_t *time_ms) { *time_ms += 51; }
+
+static void lt_mcounter_update_time(uint32_t *time_ms) { *time_ms += 51; }
+
 static void lt_r_mem_data_erase_time(uint32_t *time_ms) { *time_ms += 55; }
 
+static uint32_t change_pin_counter_cached = 0;
+static bool is_change_pin_counter_cached = false;
+
+static bool get_change_pin_counter(uint32_t *change_pin_counter) {
+  tropic_driver_t *drv = &g_tropic_driver;
+
+  if (is_change_pin_counter_cached) {
+    *change_pin_counter = change_pin_counter_cached;
+    return true;
+  }
+
+  lt_ret_t ret = lt_mcounter_get(&drv->handle, TROPIC_CHANGE_COUNTER_SLOT,
+                                 change_pin_counter);
+  if (ret == LT_OK) {
+    *change_pin_counter =
+        TROPIC_CHANGE_COUNTER_SLOT_MAX_VALUE - *change_pin_counter;
+  } else if (ret == LT_L3_COUNTER_INVALID) {
+    // The counter has not been initialized yet
+    *change_pin_counter = 0;
+  } else {
+    return false;
+  }
+
+  change_pin_counter_cached = *change_pin_counter;
+  is_change_pin_counter_cached = true;
+
+  return true;
+}
+
+static void get_change_pin_counter_time(uint32_t *time_ms,
+                                        bool is_change_pin_counter_cached) {
```

Comment:
Local variable `is_change_pin_counter_cached` shadows the global variable of the same name.

Reply:
I meant not good code style. It can lead to misinterpretation and programming errors in the future. https://wiki.sei.cmu.edu/confluence/display/c/DCL01-C.+Do+not+reuse+variable+names+in+subscopes

BTW, I'd like us to adopt a code-style policy where global variables would be prefixed with `g_`, which would have avoided this in the first place.